### PR TITLE
SW-5125: FE - Full Module View (Post sync follow-ups)

### PR DESCRIPTION
**Pull request by joshbuchea**
This PR includes various follow-up changes after the recent product sync:

- Update module types
- Refresh module & module event details automatically (once per minute)
- Deliverable due date color is adjusted based on current date
- Join session button is disabled until 15 minutes before the event
- Join session button is replaced with message after the event
- Add Module Additional Resources View
- Add Module Preparation Materials View

Outstanding tasks:

- [ ] Implement current module check
- [ ] Replace mock deliverables for module
- [ ] List session slides and other session resources?

### CODE REVIEW

File: src/scenes/ModulesRouter/ModuleEventView.tsx [hunk 1]
```diff
@@ -17,6 +17,8 @@ import { getLongDateTime } from 'src/utils/dateFormatter';
 
 import ModuleViewTitle from './ModuleViewTitle';
 
+const FIFTEEN_MINUTE_INTERVAL_MS = 15 * 60 * 1000;
```
Comment (constanzauanini) [inline src/scenes/ModulesRouter/ModuleEventView.tsx:20]: Why is this here and not in constants as the ONE_MINUTE_INTERVAL_MS ?

File: src/scenes/ModulesRouter/ModuleEntry.tsx [hunk 2]
```diff
@@ -2,32 +2,41 @@ import { Box, Grid, Typography, useTheme } from '@mui/material';
...
 import Button from 'src/components/common/button/Button';
 import useNavigateTo from 'src/hooks/useNavigateTo';
+import { useLocalization } from 'src/providers';
 import strings from 'src/strings';
 import { Module } from 'src/types/Module';
+import { getDateRangeString } from 'src/utils/dateFormatter';
 
 interface ModuleEntryProps {
+  index: number;
+  last?: boolean;
```
Comment (karthikbtf) [inline src/scenes/ModulesRouter/ModuleEntry.tsx:12]: should we use a last-child selector instead of passing in `last` as a prop?

File: src/scenes/ModulesRouter/ModuleEventSessionView.tsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    if (eventSession?.endTime) {
+      const endTime = new Date(eventSession.endTime);
+      const diff = endTime.getTime() - now.getTime();
+      return diff < 0;
+    }
+
+    return false;
+  }, [eventSession, now]);
+
+  // update the current time every minute
```
Comment (karthikbtf) [inline src/scenes/ModulesRouter/ModuleEventSessionView.tsx:88]: very nice 👍 
thoughts on extracting this out as a hook for reusability? useCountdown or useTimer or useClock or useDate.. maybe for later, we can get as simple or fancy here.

File: src/types/Module.ts [hunk 4]
```diff
@@ -1,30 +1,45 @@
...
+      return strings.ONE_ON_ONE_SESSION;
+    case 'Workshop':
+      return strings.WORKSHOP;
+    default:
+      return '';
+  }
+};
+
+export const getModuleNumber = (module?: Module, modules?: Module[]): string => {
+  return ((modules || []).findIndex((m) => m.id === module?.id) + 1).toString();
```
Comment (karthikbtf) [inline src/types/Module.ts:44]: handle `-1`?

File: src/scenes/ModulesRouter/ModuleEntry.tsx (hunk 2 above)
Comment (joshbuchea) [reply src/scenes/ModulesRouter/ModuleEntry.tsx:12]: Oh yeah I'm so used to not having selectors like that in mobile apps that I forget to use them sometimes, good call 👍

File: src/scenes/ModulesRouter/ModuleEventSessionView.tsx [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+import { selectModule } from 'src/redux/features/modules/modulesSelectors';
+import { selectProjectModuleList } from 'src/redux/features/modules/modulesSelectors';
+import { useAppDispatch, useAppSelector } from 'src/redux/store';
+import strings from 'src/strings';
+import { ModuleEventType, getModuleEventName } from 'src/types/Module';
+import { getLongDateTime } from 'src/utils/dateFormatter';
+
+import ModuleViewTitle from './ModuleViewTitle';
+
+const CALL_DESCRIPTION_HTML = `
```
[the remark this answers is not in the document]
Comment (joshbuchea) [reply src/scenes/ModulesRouter/ModuleEventSessionView.tsx:22]: Perhaps it should be broken up into several strings? 🤔

File: src/scenes/ModulesRouter/ModuleEventSessionView.tsx (hunk 3 above)
Comment (joshbuchea) [reply src/scenes/ModulesRouter/ModuleEventSessionView.tsx:88]: 🤔

**Review by karthikbtf — APPROVED** [review]
Thanks for the changes 🎉

**Comment by joshbuchea** [comment]
This PR has been around for a while, although there are a few small details outstanding I'm going to go ahead and merge it and I'll create a new PR for any follow-ups.

@nickgraz I know there's a decent amount of overlap between our PRs, hopefully these changes don't create too many conflicts. Feel free to modify whatever you need to in your PR to get everything playing together nicely. I'm also happy to make any changes you'd like to see in a follow-up PR.
